src/main/java/world/anhgelus/molehunt/Molehunt.java

```diff
@@ -38,17 +41,41 @@ public void onInitialize() {
 //            context.getSource().sendFeedback(() -> game.getRemainingText(), false);
 //            return Command.SINGLE_SUCCESS;
 //        }));
-        command.then(literal("moles").requires(source -> {
-            if (game == null) {
-                return false;
-            }
-            return game.isAMole(source.getPlayer());
-        }).executes(context -> {
+        command.then(literal("timer").requires(ServerCommandSource::isExecutedByPlayer).then(
+                literal("show").executes(context -> {
+                    timerVisibility.put(context.getSource().getPlayer(), true);
+                    context.getSource().sendFeedback(() -> Text.of("Showing molehunt timer"), false);
+
+                    var player = context.getSource().getPlayer();
+                    assert player != null;
+
+                    if (game == null || !game.hasStarted()) {
+                        player.networkHandler.sendPacket(new OverlayMessageS2CPacket(Text.of("§cGame has not started yet")));
+                    } else {
+                        player.networkHandler.sendPacket(new OverlayMessageS2CPacket(Text.of(game.getShortRemainingText())));
+                    }
+
+                    return Command.SINGLE_SUCCESS;
+                })
+        ).then(
+                literal("hide").executes(context -> {
+                    timerVisibility.put(context.getSource().getPlayer(), false);
+                    context.getSource().sendFeedback(() -> Text.of("Hiding molehunt timer"), false);
+                    return Command.SINGLE_SUCCESS;
+                })
+        ));
+        command.then(literal("moles").requires(source -> (game != null) && game.hasStarted() && game.isAMole(source.getPlayer())).executes(context -> {
```

Comment:
Was it tested?

Reply:
> but the commands suggestions don't refresh for that

So remove this part

Reply:
That means that the command will show for everyone

Reply:
No, remove your modification. It removes a feature: the suggestion was working before.